main/src/library/LazyList.flix

```diff
@@ -21,6 +21,20 @@ pub enum LazyList[a] {
     case LList(Lazy[LazyList[a]])
 }
 
+instance Eq[LazyList[a]] with Eq[a] {
+    pub def eq(l1: LazyList[a], l2: LazyList[a]): Bool = match (l1, l2) {
+        case (ENil, ENil) => true
+        case (ECons(x, xs), ECons(y, ys)) => if (x != y) false else xs == ys
+        case (ECons(x, xs), LCons(y, ys)) => if (x != y) false else xs == force ys
+        case (LCons(x, xs), ECons(y, ys)) => if (x != y) false else (force xs) == ys
+        case (LCons(x, xs), LCons(y, ys)) => if (x != y) false else (force xs) == force ys
+        case (LList(xs), LList(ys))       => (force xs) == force ys
```

Comment:
I assume the extra parens are required? (I think we have a bug in the compiler here, if I recall correctly.)

Reply:
Yes otherwise, it would parse it as `force (xs == force ys)`